include/odp/api/spec/sync.h

```diff
@@ -23,15 +23,24 @@ extern "C" {
  * @details
  * <b> Memory barriers </b>
  *
- * Memory barriers enforce ordering of memory load and store operations
- * specified before and after the barrier. These barriers may affect both
- * compiler optimizations and CPU out-of-order execution. All ODP
- * synchronization mechanisms (e.g. execution barriers, locks, queues, etc )
- * include all necessary memory barriers, so these calls are not needed when
- * using those. Also ODP atomic operations have memory ordered versions. These
- * explicit barriers may be needed when thread synchronization is based on
- * a non-ODP defined mechanism. Depending on the HW platform, heavy usage of
- * memory barriers may cause significant performance degradation.
+ * A memory barrier enforces the order between memory accesses (load and/or store instructions)
+ * specified (in program order) before the barrier with those specified after the barrier.
+ * A barrier may affect both compiler optimizations and CPU out-of-order execution. Depending
+ * on the HW platform and the barrier used, heavy usage of barriers may cause significant
+ * performance degradation.
+ *
+ * Application needs to use these memory barrier functions e.g. when threads synchronize
+ * directly through shared memory (without ODP defined mechanisms) or when a thread accesses
```

Comment:
The application may or may not "need" this, and also "when threads synchronize" doesn't just happen, it's something that the application is actively doing. So maybe something like "Application may use these ... functions e.g. to synchronize threads directly ... or to access memory mapped registers of a device."

Reply:
Fixed in v3. Section rewritten along those lines.